### PR TITLE
python3Packages.mat2: 0.12.1 -> 0.12.2

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://0xacab.org/jvoisin/mat2/-/blob/0.12.2/CHANGELOG.md

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 138164` at 5988f90d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>mat2 (python39Packages.mat2)</li>
    <li>metadata-cleaner</li>
    <li>python38Packages.mat2</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/python-modules/mat2/default.nix:6:3:
```
  |
6 | , fetchpatch
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '@mat2@' doesn't match anything in file 'dolphin/mat2.desktop'
Near pkgs/development/python-modules/mat2/default.nix:93:0:
```
   |
93 |     description = "A handy tool to trash your metadata";
   | ^
```
</li>
  </ul>
</details>

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 138164 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary>  <ul>    <li>mat2 (python39Packages.mat2)</li>    <li>metadata-cleaner</li>    <li>python38Packages.mat2</li>  </ul></details>

File: pkgs/development/python-modules/mat2/default.nix [hunk 1]
```diff
@@ -50,18 +50,6 @@ buildPythonPackage rec {
     ./executable-name.patch
     # hardcode path to mat2 executable
     ./tests.patch
-    # remove for next release
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mat2/default.nix:54]: Please remove the unused fetchpatch from inputs.

Comment (dotlambda) [reply pkgs/development/python-modules/mat2/default.nix:54]: https://github.com/NixOS/nixpkgs/pull/138196

**Pull request merged**